providers/legacyprov.c

```diff
@@ -185,6 +202,48 @@ int OSSL_provider_init(const OSSL_CORE_HANDLE *handle,
                        void **provctx)
 {
     OSSL_LIB_CTX *libctx = NULL;
+#ifndef STATIC_LEGACY
+    const OSSL_DISPATCH *tmp;
+#endif
+
+#ifndef STATIC_LEGACY
+    for (tmp = in; tmp->function_id != 0; tmp++) {
+        /*
+         * We do not support the scenario of an application linked against
+         * multiple versions of libcrypto (e.g. one static and one dynamic),
+         * but sharing a single legacy.so. We do a simple sanity check here.
+         */
+#define set_func(c, f)                          \
+        if (c == NULL)                          \
+            c = f;                              \
+        else                                    \
+            if (c != f) {                       \
+                fprintf(stderr, "FOO" #f);      \
```

Comment:
if that PR is to stay maybe use a better message here?

Reply:
Oh, that's not meant to be there at all, that's just a debug print I use for myself.  So that's going away...

Reply:
Fixed